sellable/src/execute.rs

```diff
@@ -58,7 +59,75 @@ where
         Ok(Response::new().add_attribute("method", "list"))
     }
 
-    pub fn try_buy(&mut self, deps: &mut DepsMut, info: MessageInfo) -> Result<Response, ContractError> {
+    pub fn try_buy_token(
+        &mut self,
+        deps: &mut DepsMut,
+        info: MessageInfo,
+        token_id: String,
+    ) -> Result<Response, ContractError> {
+        // check if enough fee was sent
+        if info.funds.len() == 0 {
+            return Err(ContractError::NoFundsPresent);
+        } else if info.funds.len() > 1 {
```

Comment:
We can make this slightly simpler for ourselves using pattern matching

```rust
match info.funds.as_slice() {
  [coin] => { /* this is the branch in which there is a single item */ }
  [] => { /* this is the branch where there are no items */ }
  _ => { /* this is the branch where there are multiple items */ }
}
```

The advantage of this approach is that we've already destructured our vector in the happy path and can directly address `coin`

Reply:
this is lovely logic